app.js

```diff
@@ -1,46 +1,138 @@
+        if (error !== null) {
+            console.log(error);
+        }
+    });
+}
+
+/**
+ * Writes a base64 string for a PNG file to disk as a JPEG file
+ * @param {string} base64String - base64 string of PNG file
+ */
+function writePngToDisk(base64String) {
+    const buffer = new Buffer(base64String.split(/,\s*/)[1], "base64");
+    // using middleware to change image from PNG to JPEG
+    pngToJpeg({quality: 90})(buffer).then(function (output) {
+        // writing file to disk as "out.jpeg"
+        fs.writeFile("out.jpeg", output, function (error) {
+            if (error !== null) {
+                console.log(error);
+            }
+        });
+    });
+}
+
+/**
+ * Confirms the file type of a base64 string, whether PNG, JPEG, or other
+ * @param {string} base64String - base64 string of file
+ * @returns {string} fileType - assignment of file type, which will either be PNG, JPEG, or undefined
+ */
+function checkFileType(base64String) {
+    let fileType = base64String.substring(0, 15); // getting the data/image type label
+    if (fileType.includes("png")) {
+        return "PNG"; // image is a PNG
+    } else if (fileType.includes("jpeg")) {
+        return "JPEG"; // image is a JPEG
+    } else {
+        return "undefined"; // image is not PNG or JPEG, so set as "undefined," because it cannot be used
+    }
+}
 
 /**
  * Parses the output of the TensorFlow trained model script (classify_image.py) to a JSON object
  * @param {string} result - Output from classify_image.py
  */
 function parseTensorFlowResult(result) {
-    let resultLines = result.toString().split("\n");
-    let resultObject = {"image_labels": []};
+    let sloth = false; // true if image contains a sloth
+    let resultLines = result.toString().split("\n"); // splitting up TensorFlow's result by newline
+    let resultObject = {"image_labels": [], "sloth_check": []}; // JSON object that will hold all the data
+    // iterating through each line from TensorFlow's result to fill up the JSON object
     resultLines.forEach(function (line) {
         if (line !== null && line !== undefined && line !== "") {
-            let fields = line.split("(");
-            return resultObject.image_labels.push({
-                "name": fields[0].replace(/\s/g, ""),
-                "score": fields[1].replace(/[^\d.]/g, "")
+            let fields = line.split("("); // splits "names" from "score"
+            resultObject.image_labels.push({
+                "name": fields[0].trim(), // remove trailing whitespaces
+                "score": fields[1].replace(/[^\d.]/g, "") // remove all except numbers and decimal points
             });
+            // check if the line contains a label for sloth with confidence higher than 70%
+            if(fields[0].includes("sloth") && !fields[0].includes("Ursus ursinus")) {
+                if(fields[1].replace(/[^\d.]/g, "") > 0.70) {
+                    sloth = true;
+                }
+            }
```

Comment:
You should definitely have unit tests for this logic. There may be edge cases you're not handling